source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -106,12 +106,19 @@ namespace PlayFab
             }
 
             // pipeline failed to intake the event, create a response
-            std::shared_ptr<const PlayFabEmitEventRequest> playFabEmitRequest = std::dynamic_pointer_cast<const PlayFabEmitEventRequest>(request);
             auto playFabEmitEventResponse = std::make_shared<PlayFabEmitEventResponse>();
             playFabEmitEventResponse->emitEventResult = emitResult;
 
+            // TODO: since we are std::move-ing the request,
+            //  we actually lose this info to return to the user.
+            // Do users care about this object?
+            // the entire point of this callback would be for users to retry if something went wrong,
+            // but with this particular change they must re-construct the request
+            // ( BUT which request? we no longer know...)
```

Comment:
this seems like a substantial open question and I have concerns about checking in changes while large open questions are still outstanding. What is the benefit of this change that is worth checking in with open questions like this.